Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -14183,7 +14182,8 @@ void WebPageProxy::setURLSchemeHandlerForScheme(Ref<WebURLSchemeHandler>&& handl
     ASSERT_UNUSED(handlerIdentifierResult, handlerIdentifierResult.isNewEntry);
 
     WebCore::LegacySchemeRegistry::registerURLSchemeAsHandledBySchemeHandler(scheme);
-    send(Messages::WebPage::RegisterURLSchemeHandler(handlerIdentifier, canonicalizedScheme.value()));
+    if (hasRunningProcess())
```

Comment:
If we stop sending this message, won't it break URLSchemeHandlers in some cases?  Couldn't we instead move the call to WKWebView._initializeWithConfiguration with the other calls to setURLSchemeHandlerForScheme?

Reply:
The commit log mentions that they will get sent via `WebPageCreationParameters` later on.

Reply:
I read the code flow again and here is my understanding.
- on constructor of WebPageProxy, either WebProcess is connected or not.
  - if connected, registered scheme handler is sent via `WebPage::RegisterURLSchemeHandler` immediately.
  - if not, it just stored in `m_urlSchemeHandlersByScheme`.
    - These values will be passed to WebProcess via `WebProcess::CreateWebPage`.
In `launchProcess()`,
- `m_hasRunningProcess` set to true
- `finishAttachingToWebProcess()` is called and `WebProcess::CreateWebPage` is invoked in it.

So I believe there's no chance to mis delivering scheme handler information even when they've registered in between WebPageProxy creation and WebProcess launch.

Reply:
ok